clients/rust/client/src/error.rs

```diff
@@ -0,0 +1,45 @@
+/// An error that can occur within the library.
+#[derive(Debug, thiserror::Error)]
+pub enum GammaError {
+    /// The state received from gamma did not match the state we sent to them.
+    #[error("State sent to gamma did not match what we got in the callback")]
+    GammaStateMissmatch,
+    /// The callback from gamma did not contain the required code query parameter.
+    #[error("The callback from gamma did not contain the expected code query parameter")]
+    NoCodeReceived,
+    /// Failed sending a request to gamma.
+    #[error("Failed sending request to gamma in context {context:?} due to error: {err:?}")]
+    FailedSendingRequest {
+        /// Which endpoint the request failed for.
+        context: String,
+        /// The error that occurred.
+        err: reqwest::Error,
+    },
```

Comment:
Not sure if you thougt about this, but instead of appending `: {err:?}` to the format string, it's common to instead add the `#[source]` or `#[from]` attribute to the source error field. That will usually give nicer looking backtraces.

```suggestion
    #[error("Failed sending request to gamma in context {context:?} due to error.")]
    FailedSendingRequest {
        /// Which endpoint the request failed for.
        context: String,
        /// The error that occurred.
        #[source]
        err: reqwest::Error,
    },
```

Reply:
I knew about the #[from] but not the #[source], sorta forgot about it when I started adding the custom attributes.